src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -125,6 +134,53 @@ impl CloudHypervisorInner {
         Ok(())
     }
 
+    async fn handle_block_device(
+        &mut self,
+        path: &str,
+        _id: &str,
+        read_only: bool,
+        _no_drop: bool,
+    ) -> Result<()> {
+        let socket = self
+            .api_socket
+            .as_ref()
+            .ok_or("missing socket")
+            .map_err(|e| anyhow!(e))?;
+
+        /*let num_queues: usize = if cfg.queue_num > 0 {
+            cfg.queue_num as usize
+        } else {
+            1
+        };
+
+        let queue_size: u16 = if cfg.queue_num > 0 {
+            u16::try_from(cfg.queue_size)?
+        } else {
+            1024
+        };*/
```

Comment:
I'd use the commented out code but add a couple of `const`'s for the default values:

```rust
const DEFAULT_DISK_QUEUES: usize = 1;

const DEFAULT_DISK_QUEUE_SIZE: u16 = 1024;
```

Reply:
Neither BlockConfig or BlockDevice have `queue_size` or `queue_num` so I removed this and will rely on the default value. Is there somewhere else I should be pulling the values from?

Reply:
Hi @cmaf,

> Is there somewhere else I should be pulling the values from?

Not yet. I've introduced consts like this for #6448 but not ready to raise a PR for that yet.

> Neither BlockConfig or BlockDevice have `queue_size` or `queue_num` so I removed this and will rely on the default value.

Alas, you can't rely on the default value for `num_queues` (nor probably `queue_size`) when you create the CH object since, for example:

- CH doesn't provide a client consumable version of `VmConfig` and all the objects it contains (which is a real shame).
- The actual CH `VmConfig` implementation sets custom handlers for various fields *iff* they are not included in the serialised data.
- I think we are including `num_queues` and `queue_size` in all serialised data.
- Since we can't consume any provided version of `VmConfig`, we have to define our own. But we're doing this without the extra baggage of custom handlers meaning all integer values are set to zero.
- Any `num_queues` field we create gets set to zero. This seems reasonable, but isn't a valid value when passed to CH: it must be `>=1`.

Reply:
Hi @jodh-intel @cmaf, 
Currently, no configurable `num_queues` and `queue_size` supported in `BlockConfig` or `BlockDevice`. 
It seems that it is not flexible though we can set the default value to them in vmm.
But I think we  should add the two fields:`num_queues` and `queue_size` in `BlockConfig` or `BlockDevice`, and set a default value for each of them at the same time.
If so, clh or dragonball, even later qemu, will be able to do config it as it does. 

@Tim-0731-Hzt What's your opinion about it ?

Reply:
I agree, we can do this